core/src/loader.rs

```diff
@@ -2711,4 +2744,96 @@ impl<'gc> Loader<'gc> {
             })
         })
     }
+
+    /// Loader to process callbacks for a multi-file selection dialog
+    pub fn multi_file_dialog_loader(
+        &mut self,
+        player: Weak<Mutex<Player>>,
+        dialog: DialogResultFuture,
+    ) -> OwnedFuture<(), Error> {
+        let handle = match self {
+            Loader::MultiFileDialog { self_handle, .. } => {
+                self_handle.expect("Loader not self-introduced")
+            }
+            _ => return Box::pin(async { Err(Error::NotMultiFileDialogLoader) }),
+        };
+
+        let player = player
+            .upgrade()
+            .expect("Could not upgrade weak reference to player");
+
+        Box::pin(async move {
+            let dialog_result = dialog.await;
+
+            // Dialog is done, allow opening new dialogs
+            player.lock().unwrap().ui_mut().close_file_dialog();
+
+            // Fire the load handler.
+            player.lock().unwrap().update(|uc| -> Result<(), Error> {
+                let loader = uc.load_manager.get_loader(handle);
+                let target_object = match loader {
+                    Some(&Loader::MultiFileDialog { target_object, .. }) => target_object,
+                    None => return Err(Error::Cancelled),
+                    _ => return Err(Error::NotMultiFileDialogLoader),
+                };
+
+                let mut activation = Activation::from_stub(
+                    uc.reborrow(),
+                    ActivationIdentifier::root("[File Dialog]"),
+                );
+
+                match dialog_result {
+                    Ok(dialog_result) => match dialog_result {
+                        FileDialogResult::Selection(selection) => {
+                            let mut values = Vec::with_capacity(selection.file_count().into());
+                            for ii in 0..(selection.file_count().into()) {
```

Comment:
nit: `ii` -> `i`? Just a typo? I'm not seeing `i` used in a larger scope that this would shadow.

Reply:
Just a preference for less typo-able identifier. Fixed along with removing that unwrap